boot/bootutil/src/swap_scratch.c

```diff
@@ -179,7 +179,7 @@ boot_slots_compatible(struct boot_loader_state *state)
     size_t num_sectors_secondary;
     size_t sz0, sz1;
     size_t primary_slot_sz, secondary_slot_sz;
-#ifndef MCUBOOT_OVERWRITE_ONLY
+#if !defined(MCUBOOT_OVERWRITE_ONLY) && !defined(MCUBOOT_DIRECT_XIP)
```

Comment:
Could instead create a new define, eg, `#define NO_SWAP (defined(MCUBOOT_OVERWRITE_ONLY) || defined(MCUBOOT_DIRECT_XIP))` and use through out. Or create a new `swap_none.c` file and specialize it for those cases.

Reply:
I updated the PR so that in the end a new define was not needed.